### PR TITLE
Converter script refactoring

### PR DESCRIPTION
**Pull request by skateman**
* Use rugged clone instead of calling GitHub API
* Read mixins from SASS files
* Call the conversion tasks by using #inject
* Copy non-less files based on a config hash
* Remove comments and withespace from the final toplevel file

### CODE REVIEW

**Comment by skateman** [comment]
@awood could you please review this?

**Comment by awood** [comment]
@skateman I will take a look today.  Sorry for the delay.  Thunderbird has been marking a lot of github's emails as spam. :frowning:

File: tasks/converter.rb [hunk 1]
```diff
@@ -1,444 +1,293 @@
...
-    end
-
-    def remove_xforms(transforms, *rejects)
-      transforms.reject do |xform|
-        rejects.include?(xform)
-      end
-    end
+require "#{BOOTSTRAP_GEM_ROOT}/tasks/converter/less_conversion"
+require 'rugged'
+require 'pry-byebug' # TODO
```
Comment (awood) [inline tasks/converter.rb]: Is this debug code or should this be retained?

File: tasks/converter.rb [hunk 2]
```diff
@@ -1,444 +1,293 @@
...
+      transforms.delete(:replace_spin)
+      transforms << :fix_top_level
+    when 'bootstrap-touchspin.less', 'spinner.less'
+      transforms.delete(:replace_spin)
     end
 
-    def get_tree(sha)
-      get_json("https://api.github.com/repos/#@repo/git/trees/#{sha}")
-    end
+    transforms.inject(input) { |a, e| send(e, a) }
```
Comment (awood) [inline tasks/converter.rb]: Very nice.  Clean and concise.

File: tasks/converter.rb (hunk 1 above)
Comment (skateman) [inline tasks/converter.rb]: removed

**Comment by awood** [comment]
Cuts ~150 lines out of the converter script.  Very nice.

**Comment by skateman** [comment]
@awood it also reduces network usage ;)

**Pull request merged**